sigar_windows.go

```diff
@@ -542,11 +542,103 @@ func (self *NetUdpV6ConnList) Get() error {
 }
 
 func (self *NetProtoV4Stats) Get() error {
-	return notImplemented()
+	protoV4Queries := []string{
+		`\TCPv4\Connections Active`,
+		`\TCPv4\Connections Passive`,
+		`\TCPv4\Connection Failures`,
+		`\TCPv4\Connections Reset`,
+		`\TCPv4\Connections Established`,
+
+		`\UDPv4\Datagrams Received Errors`,
+
+		`\IPv4\Datagrams Received Header Errors`,
+		`\IPv4\Datagrams Received Address Errors`,
+		`\IPv4\Datagrams Received Discarded`,
+		`\IPv4\Datagrams Received Unknown Protocol`,
+		`\IPv4\Datagrams Outbound Discarded`,
+		`\IPv4\Datagrams Outbound No Route`,
+
+		`\ICMP\Messages Received Errors`,
+		`\ICMP\Received Dest. Unreachable`,
+		`\ICMP\Messages Outbound Errors`,
+		`\ICMP\Sent Destination Unreachable`,
+	}
+
+	results, err := runRawPdhQueries(protoV4Queries)
+	if err != nil {
+		return err
+	}
+	self.TCP.ActiveOpens, results = uint64(results[0]), results[1:]
```

Comment:
Yep, you're exactly correct. Slices are only views into the same backing array. The idiom I'm using here is basically an array shift, popping the first element off each time. There is a bit more overhead than just array indexes, but those are really error prone and difficult to work with.

If `results` doesn't have enough elements, the method will panic. So its important that the struct setters and the queries are in the correct order and have the same number of elements.